drivers/src/drv8825.cpp

```diff
@@ -1,27 +1,19 @@
-#include "../include/drv8825.hpp"
+#include <drv8825.hpp>
 #include <libhal/output_pin.hpp>
 
 using namespace std::chrono_literals;
 
 
 namespace sjsu::drivers {
-drv8825::drv8825(
-    hal::output_pin& p_direction_pin, 
-    hal::output_pin& p_step_pin, 
-    hal::steady_clock& p_steady_clock, 
-    step_factor p_step_factor, 
-    int p_steps_per_rotation, 
-    hal::time_duration p_step_half_period,
-    std::array<hal::output_pin*,3> p_mode_pins) 
+drv8825::drv8825(ctor_params const& p_params)    
+    : m_direction_pin(p_params.direction_pin), 
+    m_step_pin(p_params.step_pin), 
+    m_clock(p_params.steady_clock) {
```

Comment:
Initialize all of the member variables in the initializer list. For those that you want to compute in the constructor, like `m_partial_steps_to_deg` default initialize the variable to 0 then assign in the ctro.